### PR TITLE
Remove uses of `StringPiece` and its variants

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
Upstream is set to deprecated, and delete all occurrences of `base::StringPiece`, and replace it with `std::string_view`, and according correspondent types.

This change removes the use of `base::StringPiece`, and of the headers as well across the Brave codebase.

Resolves https://github.com/brave/brave-browser/issues/33164

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
Ads changes LGTM++

**Review by tmancey — APPROVED** [review]
Ads changes LGTM++

**Review by emerick — APPROVED** [review]
Rewards changes LGTM

**Review by yrliou — APPROVED** [review]
wallet-core++ with nits.

File: components/brave_wallet/browser/keyring_service.cc [hunk 1]
```diff
@@ -232,7 +233,7 @@ absl::optional<uint32_t> ExtractAccountIndex(mojom::KeyringId keyring_id,
   return result;
 }
 
-static base::span<const uint8_t> ToSpan(base::StringPiece sp) {
+static base::span<const uint8_t> ToSpan(std::string_view sp) {
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply components/brave_wallet/browser/keyring_service.cc]: Done.

File: components/brave_wallet/browser/password_encryptor_unittest.cc [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
 #include "brave/components/brave_wallet/browser/password_encryptor.h"
-#include "base/strings/string_piece.h"
 #include "crypto/aead.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
 namespace brave_wallet {
 
 namespace {
-base::span<const uint8_t> ToSpan(base::StringPiece sp) {
+base::span<const uint8_t> ToSpan(std::string_view sp) {
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply components/brave_wallet/browser/password_encryptor_unittest.cc]: Done

**Review by fmarier — APPROVED** [review]
`chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc` changes look good.

**Comment by diracdeltas** [comment]
Please consider adding a check for this to the security action if you want to make sure nobody uses this API going forward: https://github.com/brave/security-action/tree/main/assets/semgrep_rules/client

**Review by StephenHeaps — APPROVED** [review]
iOS Wallet++

**Review by simonhong — APPROVED** [review]

**Review by AlexeyBarabash — APPROVED** [review]
sync++

**Review by goodov — APPROVED** [review]

**Review by DJAndries — APPROVED** [review]
p3a ++

**Comment by bsclifton** [comment]
Admin merge as @iefremov is PTO and this has passing CI and lots of reviews 😄

**Pull request merged**